integration_tests/contracts/contracts/TestICA.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.6;
+
+contract TestICA {
+    address constant icaContract = 0x0000000000000000000000000000000000000066;
+
+    function encodeRegister(string memory connectionID) internal view returns (bytes memory) {
+        return abi.encodeWithSignature(
+            "registerAccount(string,string)",
+            connectionID, msg.sender, ""
+        );
+    }
+
+    function callRegister(string memory connectionID) public returns (bool) {
+        (bool result, ) = icaContract.call(encodeRegister(connectionID));
+        require(result, "call failed");
+        return true;
+    }
+
+    function delegateRegister(string memory connectionID) public returns (bool) {
+        (bool result,) = icaContract.delegatecall(encodeRegister(connectionID));
+        require(result, "call failed");
+        return true;
+    }
+
+    function staticRegister(string memory connectionID) public returns (bool) {
+        (bool result,) = icaContract.staticcall(encodeRegister(connectionID));
+        require(result, "call failed");
+        return true;
+    }
+
+    function encodeQueryAccount(string memory connectionID, address addr) internal view returns (bytes memory) {
+        return abi.encodeWithSignature(
+            "queryAccount(string,address)",
+            connectionID, addr
+        );
+    }
+
+    function callQueryAccount(string memory connectionID, address addr) public returns (string memory) {
+        (bool result, bytes memory data) = icaContract.call(encodeQueryAccount(connectionID, addr));
```

Comment:
Similarly here, this appears to be a read only call but instead it is using `call`.